src/importlinter/contracts/layers.py

```diff
@@ -15,14 +15,19 @@
 from ._common import DetailedChain, build_detailed_chain_from_route, render_chain_data
 
 
+@dataclass(frozen=True)
+class Level:
+    layers: set[Layer]
+
+
 @dataclass(frozen=True)
 class Layer:
     name: str
     is_optional: bool = False
 
 
-class LayerField(fields.Field):
-    def parse(self, raw_data: str | list) -> Layer | set[Layer]:
+class LevelField(fields.Field):
```

Comment:
Something to be aware of when making changes to the fields is it is possible that someone is using them for a [custom contract](https://import-linter.readthedocs.io/en/stable/custom_contract_types.html). So this could be a breaking change, in theory.

In practice I'm not sure how much custom contracts are used, but ideally changes should be backwards-compatible. I'm open to relaxing this in certain cases though. Probably one for discussion.

Reply:
Good point. I would guess that custom contracts aren't used so much, and when they are used I'm not sure if people are importing from the existing contracts to extend them or just starting from scratch. For me this breaking change is okay - if someone really wants the old `LayerField` then they can just copy-paste those small number of lines from the old version (very low effort). It's also only a linting tool, so will only break a CI pipeline at worst (not a real running app). What do you think?